src/utils.js

```diff
@@ -672,6 +674,32 @@ function getCompileFn(implementation, options) {
       };
     }
 
+    if (options.api === "modern-compiler") {
+      return async (sassOptions) => {
+        // eslint-disable-next-line no-underscore-dangle
+        const webpackCompiler = loaderContext._compiler;
+        const { data, ...rest } = sassOptions;
+
+        // Some people can run the loader in a multi-threading way;
+        // there is no webpack compiler object in such case.
+        if (webpackCompiler) {
+          const key = isDartSass ? "dart-sass" : "sass-embedded";
+          if (!sassModernCompilers[key]) {
+            // Create a long-running compiler process that can be reused
+            // for compiling individual files.
+            const compiler = await implementation.initAsyncCompiler();
+            webpackCompiler.hooks.shutdown.tap("sass-loader", () => {
+              compiler.dispose();
```

Comment:
In my debugging this hook was never called; not sure when it should be. In any case, should we `delete sassModernCompilers[key]` after disposing? If webpack may reuse the `sass-loader` module after the `shutdown` phase, we should, otherwise the compiler won't be recreated on the next run.

Reply:
After `shutdown` should nothing happend in theory, weird, it should happend